### PR TITLE
Fix regex skipping over all src except the last

### PR DESCRIPTION
**Pull request by vaxr**
See #2918

Adapting HTML_MEDIA_TAGS to allow for `>` inside '' and "" led to multiple images inside a field sometimes being disregarded and marked as "unused". This seems to have been caused by a missing lazy (?) quantifier for the regex part skipping over non-`>`.

@meliache Thanks for finding this! If you can, please check if this resolves the bug from your end.

### CODE REVIEW

**Comment by vaxr** [comment]
Not sure why the build failed. Seems to be unrelated? Tests run locally.

File: rslib/src/media/check.rs [hunk 1]
```diff
@@ -816,6 +816,28 @@ Unused: unused.jpg
...
+        let field = "<img alt=\"foo\" src=\"foo-dd.jpg\"><img alt=\"bar\" src=\"bar-dd.jpg\">";
+        let seen = normalize_and_maybe_rename_files_helper(&mut checker, field);
+        assert!(seen.contains("foo-dd.jpg"));
+        assert!(seen.contains("bar-dd.jpg"));
+
+        let field = "<img alt='foo' src='foo-sd.jpg'><img alt=\"bar\" src=\"bar-sd.jpg\">";
+        let seen = normalize_and_maybe_rename_files_helper(&mut checker, field);
+        assert!(seen.contains("foo-sd.jpg"));
+        assert!(seen.contains("bar-sd.jpg"));
+
```
Comment (meliache) [inline rslib/src/media/check.rs:840]: Would it make sense to add some tests for where there's whitespace, a newline or text-characters between the image tags? Also maybe tests for when there are more than two images? For the particular bug that I saw that didn't matter, the bug was always the same. But in my mind ideally, the tests should cover more than the bug that we just fixed, but also unforeseen future bugs. On the other hand that would add a lot of boilerplate code. I'm haven't done much Anki development, so not sure what the testing guidelines here are...

Comment (vaxr) [reply rslib/src/media/check.rs:840]: For the issue at hand, the tests added are the minimum retention test addressing this particular fix. There are three because she bug only showed up on subsequent usage of single or double quotes. That's all a retention test should do as far as I'm concerned.

I agree it would be great to have more tests in general. How detailed the tests should be is a matter of debate I suppose. Surely there are lower hanging fruits.

**Comment by meliache** [comment]
> please check if this resolves the bug from your end.

Compiling finished, I can't reproduce the bug on this PR, thanks :+1:

File: rslib/src/media/check.rs (hunk 1 above)
Comment (meliache) [reply rslib/src/media/check.rs:840]: Thanks for explaining, that's fine with me.

**Comment by dae** [comment]
Thank you @meliache for spotting the problem, and thank you @vaxr for the quick fix, complete with tests!
